src/html/HTMLWriter.jl

```diff
@@ -365,6 +365,21 @@ value is `"en"`.
 **`warn_outdated`** inserts a warning if the current page is not the newest version of the
 documentation.
 
+**`size_threshold`** sets the maximum allowed HTML file size (in bytes) that Documenter is allowed to
+generate for a page. If the generated HTML file is larged than this, Documenter will throw an error and
+the build will fail. If set to `nothing`, the file sizes are not checked. Defaults to `200 KiB` (but
+increases of this default value will be considered to be non-breaking).
+
+**`size_threshold_warn`**: like `size_threshold`, but going over this limit will only cause Documenter to
+print a warning, instead of throwing an error. Defaults to `100 KiB`, and must be less than or equal to
+`size_threshold`.
+
+!!! note "Purpose of HTML size thresholds"
+
+    The size threshold, with a reasonable default, exists so that users would not deploy huge pages
+    accidentally (which among other this will result in bad UX for the readers and negatively impacts
+    SEO). It is relatively easy to have e.g. an `@example` produce a lot of output.
```

Comment:
What is the alternative? Could documenter automatically save plots to file and load them from there?

Reply:
JuMP has quite a few pages which are large:

 * (292 kb) https://github.com/jump-dev/JuMP.jl/blob/gh-pages/v1.13.0/tutorials/applications/power_systems/index.html
 * (279 kb) https://github.com/jump-dev/JuMP.jl/blob/gh-pages/v1.13.0/tutorials/getting_started/getting_started_with_data_and_plotting/index.html
 * (438 kb) https://github.com/jump-dev/JuMP.jl/blob/gh-pages/v1.13.0/tutorials/nonlinear/space_shuttle_reentry_trajectory/index.html

Reply:
Good point, I should also check JuMP docs. But yes: #2143. It won't be in 1.0, but hopefully in 1.1 or so. In the meanwhile, the solution is just to set `size_threshold` large enough (or to `nothing`).

Reply:
What if there was an `exclude::Vector{String}` argument where you could explicitly ignore some large pages?

Reply:
That seems like a good idea. Forcing people to set a global size setting to ignore one page is not ideal. I'll merge this as is, but let's get that into 1.0 in a follow-up PR.